test/e2e/targets/azureeventhubs/main.go

```diff
@@ -0,0 +1,278 @@
+	"github.com/triggermesh/triggermesh/test/e2e/framework/ducktypes"
+)
+
+var targetAPIVersion = schema.GroupVersion{
+	Group:   "targets.triggermesh.io",
+	Version: "v1alpha1",
+}
+
+const (
+	targetKind = "AzureEventHubsTarget"
+)
+
+/*
+  This test requires:
+  - Azure Service Principal Credentials with the Azure Event Hubs Data Owner role assigned at the subscription level
+  The following environment variables _MUST_ be set:
+  - AZURE_SUBSCRIPTION_ID - Common subscription for the test to run against
+  - AZURE_TENANT_ID - Azure tenant to create the resources against
+  - AZURE_CLIENT_ID - The Azure ServicePrincipal Client ID
+  - AZURE_CLIENT_SECRET - The Azure ServicePrincipal Client Secret
+  - AZURE_REGION - Define the Azure region to run the test (default westus2)
+
+  These will be done by the e2e test:
+  - Create an Azure Resource Group, EventHubs Namespace, and EventHub
+  - Send an event to the Azure EventHub via the AzureEventHub Target and verify the event was sent
+*/
+
+var _ = FDescribe("Azure EventHubs target", func() {
+	ctx := context.Background()
+	subscriptionID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+	region := os.Getenv("AZURE_REGION")
+
+	if region == "" {
+		region = "westus2"
+	}
+
+	f := framework.New("azureeventhubstarget")
+
+	var ns string
+	var tgtURL *url.URL
+	var tgtClient dynamic.ResourceInterface
+
+	var rg armresources.ResourceGroup
+	var hub *eventhubs.Hub
+
+	BeforeEach(func() {
+		ns = f.UniqueName
+
+		gvr := targetAPIVersion.WithResource("azureeventhubstargets")
+		tgtClient = f.DynamicClient.Resource(gvr).Namespace(ns)
+	})
+
+	Context("a target is deployed", func() {
+		BeforeEach(func() {
+			By("creating an azure resource group", func() {
+				rg = azure.CreateResourceGroup(ctx, subscriptionID, ns, region)
+				DeferCleanup(func() {
+					By("deleting the azure resource group", func() {
+						_ = azure.DeleteResourceGroup(ctx, subscriptionID, *rg.Name)
+					})
+				})
+			})
+
+			By("creating an azure eventhub", func() {
+				hub = azure.CreateEventHubComponents(ctx, subscriptionID, ns, region, *rg.Name)
+			})
+		})
+
+		When("the spec contains default settings", func() {
+			var event *cloudevents.Event
+
+			BeforeEach(func() {
+				By("creating an AzureEventHubTarget object", func() {
+					tgt, err := createTarget(tgtClient, ns, "test-",
+						withServicePrincipal(),
+						withEventHubID(subscriptionID, *rg.Name, ns, ns))
+
+					Expect(err).ToNot(HaveOccurred())
+
+					tgt = ducktypes.WaitUntilReady(f.DynamicClient, tgt)
+					tgtURL = ducktypes.Address(tgt)
+					Expect(tgtURL).ToNot(BeNil())
+				})
+			})
+
+			It("receives an event on the Event Bus", func() {
+				var payload []byte
+				var partitionIDs []string
+				var eventHandler eventhubs.Handler
+				eventReceivedChannel := make(chan bool)
+				var evCtx context.Context       // Used to set a timeout for reading events
+				var evCancel context.CancelFunc // Used to cancel the reading context
```

Comment:
Most of the variables above do not need to carry between spec/by blocks, and shouldn't be declared here. They confuse the reader about what needs to be in the Ginkgo scope, and what doesn't 
Don't hesitate to declare these inside By() blocks, this is exactly where they belong.

Reply:
With the exception of `payload`, the rest are used by multiple `By()` blocks which is why they are located where they are.